crates/sui/src/upgrade_compatibility.rs

```diff
@@ -1509,6 +1545,65 @@ fn enum_ability_mismatch_diag(
     Ok(diags)
 }
 
+fn enum_variant_field_error(
+    old_variant: &Variant,
+    new_variant: &Variant,
+    variant_loc: Loc,
+    def_loc: Loc,
+) -> (DiagnosticInfo, Vec<String>) {
+    if old_variant.fields.len() != new_variant.fields.len() {
+        return (
+            Declarations::FieldMismatch.into(),
+            vec![format!(
+                "Mismatched variant field count, expected {}, found {}.",
```

Comment:
nit (ignore if not relevant): most places we do `message {}, expected {}`. Here it is reversed: expected, found.